packages/rules/src/nestedWorkspaces.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright 2019 Palantir Technologies, Inc.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ *
+ */
+
+import { Context, RuleModule } from "@monorepolint/core";
+import globby from "globby";
+import * as r from "runtypes";
+
+export const Options = r.Undefined;
+
+type Options = r.Static<typeof Options>;
+
+// Custom monorepolint rule to enforce that the root package.json contains
+// all of the workspaces in the repo (including nested packages)
+export const nestedWorkspaces: RuleModule<typeof Options> = {
+  check: async (context: Context) => {
+    const rootPackageJson = context.getWorkspaceContext().getPackageJson();
+
+    // Expand a set of globs covering all package.json files in the entire repo (except the root)
+    const packageJsonPaths = await globby(["*/**/package.json", "!**/node_modules/**"]);
+
+    const workspaces = Array.isArray(rootPackageJson.workspaces)
+      ? rootPackageJson.workspaces
+      : rootPackageJson.workspaces !== undefined
```

Comment:
oh sad, monorepolint is behind on typescript

Reply:
Yeah I wrote it so much nicer with optional chaining, but then was overcome by sadness when I realised it was on 3.5 :(